includes/discovery/bgp-peers/dell-os10.inc.php

```diff
@@ -52,7 +54,7 @@
                 // Setting it here avoids the code that resets it to null if not found in BGP4-MIB.
                 $bgpLocalAs = $value['os10bgp4V2PeerLocalAs'];
 
-                if (dbFetchCell('SELECT count(*) FROM `bgpPeers` WHERE device_id = ? AND bgpPeerIdentifier = ? AND vrf_id = ?', [$device['device_id'], $address, $vrfId]) == 0) {
+                if (BgpPeer::where('device_id', '=', $device['device_id'])->where('bgpPeerIdentifier', '=', $address)->where('vrf_id', '=', $vrfId)->count() < 1) {
```

Comment:
`! DeviceCache::getPrimary()->bgppeers()->where('bgpPeerIdentifier', '=', $address)->where('vrf_id', '=', $vrfId)->exists()`

Reply:
Better yet:

```php
BgpPeer::updateOrCreate([
    'device_id' => $device['device_id'],
    'bgpPeerIdentifier' => $address,
    'vrf_id' => $vrfId,
], $row);

```

Reply:
If it won't create because the other fields don't have values, it should be handled by default values or nullable.

Reply:
It slightly changes the behaviour for the '+' or '.' beeing typed ... But it can be achieved with an 'else' statement.

Reply:
Yes indeed.  There is also a ModuleModelObserver which you can call like this ModuleModelObserver::observe(BgpPeer::class);  which will install an event listener to output +, -, and .